pkg/networkdevice/diagnoses/diagnoses.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package diagnoses implements the diagnosis collection for NDM resources
+package diagnoses
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	"github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+)
+
+// Diagnoses hold diagnoses for a NDM resource
+type Diagnoses struct {
+	resourceType  string
+	resourceID    string
+	diagnoses     []metadata.Diagnosis
+	prevDiagnoses []metadata.Diagnosis
+}
+
+var severityMap = map[string]diagnosis.Result{
+	"success": diagnosis.DiagnosisSuccess,
+	"error":   diagnosis.DiagnosisFail,
+	"warn":    diagnosis.DiagnosisWarning,
+}
+
+// NewDeviceDiagnoses returns a new Diagnoses for a NDM device resource
+func NewDeviceDiagnoses(deviceID string) *Diagnoses {
+	return &Diagnoses{
+		resourceType: "ndm_device",
+		resourceID:   deviceID,
+	}
+}
+
+// Add adds a diagnoses
+func (d *Diagnoses) Add(result string, errorCode string, message string) {
+	d.diagnoses = append(d.diagnoses, metadata.Diagnosis{
+		Severity:  result,
+		ErrorCode: errorCode,
+		Diagnosis: message,
+	})
+}
+
+// ReportDiagnosis returns diagnosis metadata
+func (d *Diagnoses) ReportDiagnosis() []metadata.DiagnosisMetadata {
```

Comment:
```suggestion
func (d *Diagnoses) Report() []metadata.DiagnosisMetadata {
```

🥜 Maybe we can just use Report since it should be clear we are reporting diagnoses.